woocommerce/woocommerce-ios
# [Analytics Hub] Create shared set of stats helpers for My Store and Analytics Hub

Part of #8149

## Description

This PR moves a number of stats data helpers from the `StoreStatsPeriodViewModel` (used for My Store stats) to a separate `StatsV4DataHelper` class, so they can be reused in the Analytics Hub.

It also adds a new helper related to the new Average Order Value stat that will be used in Analytics Hub.

## Changes

* Moves the shared helpers and unit tests from `StoreStatsPeriodViewModel` to `StatsV4DataHelper`.
* Adds a new public helper `StatsV4DataHelper.createAverageOrderValueText(orderStats:currencyFormatter:currencyCode:)`.

## Testing

This PR refactors the helpers but there should be no change in the current behavior:

1. Build and run the app.
2. Confirm your stats (text and chart) appear as expected on the My Store screen.
3. Confirm you can select different time period or select specific intervals on the stats chart, and the stats text is updated as expected.

**Note:** This PR removes some redundant view model unit tests, because the logic is now contained in the helper class. However, you can undo the changes from 466c36f and run the tests in `StoreStatsPeriodViewModelTests` to confirm that all of the previous tests are still passing with these changes.

## Submitter Checklist

Update release notes:
